### ISSUE TITLE
[Warrior] Rampage randomly disappears

### ISSUE THREAD

**Issue by Centan** [FeenixServerProject/Archangel_2.4.3_Bugtracker#3639]
Desired Situation:
After casting Rampage each crit increases its stack till max 5.
From now on 5 more crits will refresh his 30s uptime. After the 5th crit it starts running out to 0s (even though there are more crits)

Current Version:
After casting Rampage each crit increases its stack till max 5. 
From now on every crit refreshes his 30s uptime (even after the 5th one).
Then randomly the effect disappears without running out. (Not sure what triggers this)

**Comment by Vinxhe** [comment]
I did some more testing on it and I think I isolated the problem.

What you need:
- lvl 70 fury warrior
- some training dummy
- stopwatch (/sw)
Hit the dummy until you crit, use Rampage and start the stopwatch in the same moment. You will notice that the Rampage buff disappears exactly when the stopwatch has 30 seconds on it.

What should happen:
You use Rampage, Rampage stacks 5 times and refreshes itself a couple times. The buff (lasts 30 seconds) then runs out, unless you manually use Rampage again to keep up the 5 stacks.

What actually happens:
You use Rampage, a 30 second timer starts. After the 30 seconds the Rampage buff disappears. It doesn't matter how many stacks of Rampage there are, or what the buff times says.

So the buff has some kind of internal timer which the player himself won't see. The actual buff timer shows the right time and refreshes correctly but doesnt actually determine the buffs uptime.

**Comment by effinstein** [comment]
Also, there's another problem with refreshing it. 

What should happen:
You have rampage up with 5 stacks running out and you want to refresh it.
You use rampage to refresh it keeps the 5 stacks and timer gets back to 30 seconds, your next 5 crits refreshing the 30 seconds duration and afterwards the the 30 seconds countdown start regardless of your crits.

What actually happens:
You have rampage up with 5 stacks and try to refresh it, knowing it will run out soon (even though at the moment every crit refreshes its duration)
You use rampage to refresh it, but rampage(with 5 stacks) disappears and reappears on your next crit with 1 stack.

So basically even if you can't actually know when it will run out you try to randomly refresh it and it actually still disappears. So something there doesn't work at all.

**Comment by devidao** [comment]
what should happen in this case?
1. You hit critcal hit
2. Rampage becomes available to cast
3. You cast rampage and get rampage buff (stack 1)
4. You hit critcal hit again
5. Rampage becomes available to cast
6. You cast rampage manually -> should you get rampage buff (stack 2) or rampage buff should disappears ??

**Comment by Siggiman** [comment]
Bump.
Still not fixed on Archangel.

**Comment by veremi** [comment]
Bump
Still not fixed on Archangel

**Comment by repae1f** [comment]
http://www.wow-one.com/forum/topic/105928-community-update/?page=2#comment-767027

**Comment by Siggiman** [comment]
Day 396 - fix still not applied on Archangel

**Comment by Siggiman** [comment]
Bump
Another patch without implementing the rampage fix on archangel :(
